pkg/compute/rust.go

```diff
@@ -170,88 +184,64 @@ func (r Rust) Verify(out io.Writer) error {
-				"To fix this error, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
-				text.Bold("Cargo.toml"),
-				text.Bold(fmt.Sprintf(`fastly = "^%s"`, latest)),
-				text.Bold("cargo update -p fastly"),
-			),
-		}
+	// Create a semver contraint to be within the latest minor range or above.
+	// TODO(phamann): Update this to major when fastly-sys hits 1.x.x.
+	fastlySysConstraint, err := semver.NewConstraint(fmt.Sprintf("~%d.%d.0", latestFastlySys.Major(), latestFastlySys.Minor()))
+	if err != nil {
+		return fmt.Errorf("error parsing latest crate version: %w", err)
 	}
 
-	// Parse lockfile version to semver.
-	version, err := semver.Parse(crate.Version)
+	fastlySysVersion, err := getCrateVersionFromMetadata(metadata, "fastly-sys")
+	// If fastly-sys crate not found, error with dual remediation steps.
 	if err != nil {
-		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+		return newCargoUpdateRemediationErr(err, latestFastly.String())
 	}
 
-	// Parse the lowest minor semver for the latest release.
-	// I.e. v0.3.2 becomes v0.3.0.
-	latestMinor, err := semver.Parse(fmt.Sprintf("%d.%d.0", latest.Major, latest.Minor))
-	if err != nil {
-		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+	// If fastly-sys version doesn't meet our constraint of being within the
+	// minor range, error with dual remediation steps.
+	if ok := fastlySysConstraint.Check(fastlySysVersion); !ok {
+		return newCargoUpdateRemediationErr(fmt.Errorf("fastly crate not up-to-date"), latestFastly.String())
 	}
 
-	// If the lockfile version is within the minor range but lower than the
-	// latest, error with remediation to run `cargo update`.
-	if version.GTE(latestMinor) && version.LT(latest) {
-		return errors.RemediationError{
-			Inner:       fmt.Errorf("fastly crate not up-to-date"),
-			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s\n", text.Bold("cargo update -p fastly")),
-		}
+	fastlyVersion, err := getCrateVersionFromMetadata(metadata, "fastly")
+	// If fastly crate not found, error with dual remediation steps.
+	if err != nil {
+		return newCargoUpdateRemediationErr(err, latestFastly.String())
 	}
 
-	// If version is on a lower minor or major, error with a dual remediation.
-	if version.LT(latest) {
-		return errors.RemediationError{
-			Inner: fmt.Errorf("fastly crate not up-to-date"),
-			Remediation: fmt.Sprintf(
-				"To fix this error, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
-				text.Bold("Cargo.toml"),
-				text.Bold(fmt.Sprintf(`fastly = "^%s"`, latest)),
-				text.Bold("cargo update -p fastly"),
-			),
-		}
+	// If fastly crate version is lower than the latest, suggest user should
+	// update, but don't error.
+	if fastlyVersion.LessThan(latestFastly) {
+		text.Break(out)
+		text.Info(out, fmt.Sprintf(
+			"a newer version of the fastly crate is avaiable, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
```

Comment:
Trying to emphasize they don't have to do this immediately if the `fastly-sys` dependency is okay.
```suggestion
			"an optional upgrade for the fastly crate is available, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
```